### PR TITLE
Cleaned MusicListView

### PR DESCRIPTION
**Pull request by Nine-H**
redid it, it's actually even better.

- renamed variables in elementary style
- renamed methods in elementary style (except a public one)
- removed two separator declarations
- grouped property changes with the associated object
- removed build_ui() method and placed it's arguments in constructor
- fixed the two submenus in the same ways.
- fixed inconsistent spacing
- changed `update_sensitivities()` to use a switch/case block

### CODE REVIEW

File: src/Views/ListView/Lists/MusicListView.vala [hunk 1]
```diff
@@ -83,110 +78,68 @@ public class Noise.MusicListView : GenericList {
...
-        mediaMenuQueue = new Gtk.MenuItem.with_label(C_("Action item (verb)", "Queue"));
-        mediaMenuAddToPlaylist = new Gtk.MenuItem.with_label(_("Add to Playlist"));
-        mediaRemove = new Gtk.MenuItem.with_label(_("Remove Song"));
-        importToLibrary = new Gtk.MenuItem.with_label(_("Import to Library"));
-        mediaRateMedia = new Granite.Widgets.RatingMenuItem ();
-
-        mediaActionMenu = new Gtk.Menu ();
-        mediaActionMenu.attach_to_widget (this, null);
-
+        button_release_event.connect(view_click_release);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 2]
```diff
@@ -83,110 +78,68 @@ public class Noise.MusicListView : GenericList {
...
+            media_action_menu.append (media_edit_media);
         }
-
-        mediaActionMenu.append(mediaFileBrowse);
-        mediaActionMenu.append(mediaMenuContractorEntry);
+        media_action_menu.append (media_file_browse);
+        media_action_menu.append (media_menu_contractor_entry);
         if (read_only == false) {
-            mediaActionMenu.append(mediaRateMedia);
+            media_action_menu.append(media_rate_media);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 3]
```diff
@@ -195,15 +148,45 @@ public class Noise.MusicListView : GenericList {
         update_sensitivities ();
     }
 
-    public void popup_media_menu (Gee.Collection<Media> selection) {
-        // Create add-to-playlist menu
-        var addToPlaylistMenu = new Gtk.Menu ();
+    public override void update_sensitivities () {
+        media_action_menu.show_all();
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 4]
```diff
@@ -195,15 +148,45 @@ public class Noise.MusicListView : GenericList {
...
 
-        var mediaMenuNewPlaylist = new Gtk.MenuItem.with_label(_("New Playlist…"));
-        mediaMenuNewPlaylist.activate.connect(mediaMenuNewPlaylistClicked);
-        addToPlaylistMenu.append (mediaMenuNewPlaylist);
+        if (hint == ViewWrapper.Hint.MUSIC) {
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: If you're touching this code, it should use a switch/case statement. See #174

File: src/Views/ListView/Lists/MusicListView.vala [hunk 5]
```diff
@@ -195,15 +148,45 @@ public class Noise.MusicListView : GenericList {
...
+            media_remove.set_label(_("Remove from Library"));
+            import_to_library.set_visible(false);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: should be `visible = false`

File: src/Views/ListView/Lists/MusicListView.vala [hunk 6]
```diff
@@ -195,15 +148,45 @@ public class Noise.MusicListView : GenericList {
...
-        // Create add-to-playlist menu
-        var addToPlaylistMenu = new Gtk.Menu ();
+    public override void update_sensitivities () {
+        media_action_menu.show_all();
 
-        var mediaMenuNewPlaylist = new Gtk.MenuItem.with_label(_("New Playlist…"));
-        mediaMenuNewPlaylist.activate.connect(mediaMenuNewPlaylistClicked);
-        addToPlaylistMenu.append (mediaMenuNewPlaylist);
+        if (hint == ViewWrapper.Hint.MUSIC) {
+            media_remove.set_label(_("Remove from Library"));
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: should be `label = _("Remove from Library")`

File: src/Views/ListView/Lists/MusicListView.vala [hunk 7]
```diff
@@ -195,15 +148,45 @@ public class Noise.MusicListView : GenericList {
...
+
+    public void popup_media_menu (Gee.Collection<Media> selection) {
+        var media_menu_new_playlist = new Gtk.MenuItem.with_label (_("New Playlist…"));
+        media_menu_new_playlist.activate.connect (media_menu_new_playlist_clicked);
+        
+        var add_to_playlist_menu = new Gtk.Menu ();
+        add_to_playlist_menu.append (media_menu_new_playlist);
         if (parent_wrapper.library.support_playlists () == false) {
-            mediaMenuNewPlaylist.set_visible(false);
+            media_menu_new_playlist.set_visible (false);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: should be `visible = false`

File: src/Views/ListView/Lists/MusicListView.vala [hunk 8]
```diff
@@ -235,13 +217,13 @@ public class Noise.MusicListView : GenericList {
         }
 
         if (temporary_count < 1) {
-            importToLibrary.set_sensitive (false);
+            import_to_library.set_sensitive (false);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: should be `sensitive = false`

File: src/Views/ListView/Lists/MusicListView.vala [hunk 9]
```diff
@@ -344,7 +326,7 @@ public class Noise.MusicListView : GenericList {
     }
 
     /* button_release_event */
-    private bool viewClickRelease(Gtk.Widget sender, Gdk.EventButton event) {
+    private bool view_click_release(Gtk.Widget sender, Gdk.EventButton event) {
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 10]
```diff
@@ -83,110 +78,68 @@ public class Noise.MusicListView : GenericList {
...
+        
+        media_menu_contractor_entry = new Gtk.MenuItem.with_label (_("Other actions"));
+        
+        media_menu_queue = new Gtk.MenuItem.with_label (C_("Action item (verb)", "Queue"));
+        media_menu_queue.activate.connect (media_menu_queue_clicked);
+        
+        media_menu_add_to_playlist = new Gtk.MenuItem.with_label (_("Add to Playlist"));
+        
+        media_remove = new Gtk.MenuItem.with_label (_("Remove Song"));
+        media_remove.activate.connect(mediaRemoveClicked);
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 11]
```diff
@@ -285,14 +267,14 @@ public class Noise.MusicListView : GenericList {
...
+            media_menu_contractor_entry.sensitive = contractorSubMenu.get_children ().length () > 0;
             contractorSubMenu.show_all ();
         } catch (Error err) {
             warning ("Failed to obtain Contractor actions: %s", err.message);
-            mediaMenuContractorEntry.sensitive = false;
+            media_menu_contractor_entry.sensitive = false;
         }
 
-        mediaActionMenu.popup (null, null, null, 3, Gtk.get_current_event_time());
+        media_action_menu.popup (null, null, null, 3, Gtk.get_current_event_time());
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: space before ()

File: src/Views/ListView/Lists/MusicListView.vala [hunk 12]
```diff
@@ -195,15 +148,52 @@ public class Noise.MusicListView : GenericList {
...
+                } else {
+                    media_remove.visible = false;
+                }
+                break;
+            case ViewWrapper.Hint.SMART_PLAYLIST:
+                media_remove.visible = false;
+                import_to_library.visible = false;
+                break;
+            case ViewWrapper.Hint.DEVICE_AUDIO:
+                 media_edit_media.visible = false;
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: there's extra space here

File: src/Views/ListView/Lists/MusicListView.vala [hunk 13]
```diff
@@ -195,15 +148,52 @@ public class Noise.MusicListView : GenericList {
...
+            case ViewWrapper.Hint.READ_ONLY_PLAYLIST:
+                import_to_library.visible = false;
+                if (playlist == App.player.queue_playlist) {
+                    media_remove.label = _("Remove from Queue");
+                    media_menu_queue.visible = false;
+                } else {
+                    media_remove.visible = false;
+                }
+                break;
+            case ViewWrapper.Hint.SMART_PLAYLIST:
```
Comment (danrabbit) [inline src/Views/ListView/Lists/MusicListView.vala]: This is the same as the default case, so you can just group it down there by `default:`

Comment (Nine-H) [reply src/Views/ListView/Lists/MusicListView.vala]: group it or remove the case?

Comment (danrabbit) [reply src/Views/ListView/Lists/MusicListView.vala]: I guess it could just be removed :)

**Pull request merged**